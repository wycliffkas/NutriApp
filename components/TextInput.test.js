import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Input from './TextInput';

describe('Input component', () => {
  const mockOnChangeText = jest.fn();
  const placeholderText = 'Test placeholder';
  const testValue = 'Test input value';

  it('renders the component correctly', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder={placeholderText} onChangeText={mockOnChangeText} />
    );
    const inputElement = getByPlaceholderText(placeholderText);
    expect(inputElement).toBeDefined();
  });

  it('triggers onChangeText when input value changes', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder={placeholderText} onChangeText={mockOnChangeText} />
    );
    const inputElement = getByPlaceholderText(placeholderText);
    fireEvent.changeText(inputElement, testValue);
    expect(mockOnChangeText).toHaveBeenCalledWith(testValue);
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <Input placeholder={placeholderText} onChangeText={mockOnChangeText} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
