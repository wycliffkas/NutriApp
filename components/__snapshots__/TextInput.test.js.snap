// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<TextInput
  onChangeText={
    [MockFunction] {
      "calls": [
        [
          "Test input value",
        ],
      ],
      "results": [
        {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  placeholder="Test placeholder"
  style={
    {
      "borderBottomColor": "black",
      "borderBottomWidth": 1,
      "fontSize": 16,
      "marginBottom": 20,
      "width": 37,
    }
  }
/>
`;
